Python/thread_pthread.h

```diff
@@ -302,6 +308,29 @@ PyThread_get_thread_ident(void)
     return (unsigned long) threadid;
 }
 
+#ifdef PY_HAVE_THREAD_NATIVE_ID
+unsigned long
+PyThread_get_thread_native_id(void)
+{
+    if (!initialized)
+        PyThread_init_thread();
+#ifdef __APPLE__
+    uint64_t native_id;
+    pthread_threadid_np(NULL, &native_id);
+#elif defined(__linux__)
+    pid_t native_id;
+    native_id = syscall(__NR_gettid);
+#elif defined(__FreeBSD__)
+    pid_t native_id;
+    native_id = pthread_getthreadid_np();
+#else
+    unsigned long native_id;
+    native_id = 0;
+#endif
+    return (unsigned long) native_id;
```

Comment:
Hum,  this downcast leads to integer overflow on macOS which uses uint64_t if unsigned long is 32-bit, no?

What about using uint64_t type for the return type?

Reply:
From the last PR: https://github.com/python/cpython/pull/11993#discussion_r283107275

>On Apple platforms, the `pthread_threadid_np()` function wasn’t introduced until Mac OS 10.5, after Mac OS primarily became 64-bit. I figured it was acceptable to have the 64-bit type here since Python 3 for Mac only builds 64-bit versions and therefore the only Mac OS versions capable of supporting this feature would have to be those that also support Python 3 and therefore be 64-bit.

Reply:
I checked my notes, Python 3.8 only supports macOS 10.6 and newer, so it's ok :-) https://pythondev.readthedocs.io/platforms.html#well-supported-platforms